cracking-the-coding-interview/problems/1.strings/1.isUnique.ts

```diff
@@ -0,0 +1,33 @@
+// 1. * Is Unique *: Implement an algorithm to determine if a string has all unique characters.
+const isUnique = (value) => {
+  if(!value) return false;
+
+  const chars = {}
+
+  for(let i = 0; i < value.length; i++) {
+    const charAtI = value.charAt(i)
+    if(chars[charAtI] !== undefined){
+      return false
+    } else {
+      chars[charAtI] = i
+    }
+  }
+  return true
+}
+
+// What if you cannot use additional data structures ?
+const isUniqueString = (value) => {
```

Comment:
The chars variable here is being used as an array. This does not conform to the prompt "do not use additional data structures".

Reply:
fixed